### ISSUE TITLE
Sending a photo with text attached causes only the text to send (on Android both the image and text appear and on macOS only the image appears)

### ISSUE THREAD

**Issue by CSab6482** [BlueBubblesApp/bluebubbles-app#617]
![Android 1](https://user-images.githubusercontent.com/53708281/103486168-d5765780-4dc9-11eb-8d0e-9c36ab256f22.jpg)
![macOS 1](https://user-images.githubusercontent.com/53708281/103486173-da3b0b80-4dc9-11eb-9a31-84e9c92230c4.jpeg)
![iOS 1](https://user-images.githubusercontent.com/53708281/103486178-e0c98300-4dc9-11eb-8f27-7f6bbda5206f.PNG)
![Android 2](https://user-images.githubusercontent.com/53708281/103486170-d6a78480-4dc9-11eb-97cd-b7c6a49abb5f.jpg)
![macOS 2](https://user-images.githubusercontent.com/53708281/103486181-e8892780-4dc9-11eb-8af4-b05a91c60a88.jpeg)
![iOS 2](https://user-images.githubusercontent.com/53708281/103486184-ecb54500-4dc9-11eb-8c9e-9694b20643d3.PNG)
![Android 3](https://user-images.githubusercontent.com/53708281/103486290-d6f44f80-4dca-11eb-9278-2e19b024f91d.jpg)
![macOS 3](https://user-images.githubusercontent.com/53708281/103486187-f474e980-4dc9-11eb-8b7c-aea50c7b1998.jpeg)
![iOS 3](https://user-images.githubusercontent.com/53708281/103486189-f6d74380-4dc9-11eb-9373-fe2e07d627a3.PNG)

I first encountered this strange bug in mid-December but since 99% of the other photos with text that I sent had no issues I thought it was just a strange one-time thing. However, I've had the same issue twice this past weekend so I then realized it was a repeated bug. This issue is present on all builds of the Android app (e.g. stable, beta, nightly). I also got the double picture bug on the third example.

**Comment by zlshames** [comment]
yeah, i've experienced this too, however, I can't get it to fail everytime, only sometimes. Will need to research to replicate a bit more

**Comment by CSab6482** [comment]
I just learned this bug also affects desktop

**Comment by zlshames** [comment]
@CSab6482 does this still happen? I coulda sworn that i did something on the server to mitigate this

**Comment by CSab6482** [comment]
I’ve gotten into a habit of sending images and text separately because of this bug, but I’ll test it throughout this week and let you know how it goes.

**Comment by tneotia** [comment]
I don't think this happens anymore as I've sent text with a pic a bunch and never seen this

**Comment by CSab6482** [comment]
Frick. I keep saying I'll test this then I never do. Ever since I got this bug I've always separated the attachments and text just to be safe. I'd say it's something that happened like once every three months when I did face it, but maybe with some forced testing I can see more details. However, with the new server changes that check for dupe messages and such, it might be fixed. I'm giving myself a hard deadline of August 20 to test this extensively. I'll give a follow up then.

**Comment by CSab6482** [comment]
Screenshots to come later, but I was in fact able to reproduce this on 1.3.0 beta 4.

**Comment by tneotia** [comment]
So if I understand correctly, it shows the text + pic on Android, text + pic on macOS iMessage, but only the text on iOS iMessage?

**Comment by mdecourcy** [comment]
@tneotia @CSab6482 I'm currently experiencing this issue now. m1 mac mini as a server, google pixel 6 running graphene os client

**Comment by CSab6482** [comment]
> m1 mac mini as a server

on what version of macOS?

> graphene OS client

Since this app uses Google's FireBase, I'm curious to know if notifications work for you. Are you utilizing microG or anything similar?

**Comment by mdecourcy** [comment]
@CSab6482

m1 is running Monterey, 12.1

For graphene, they offer sandboxed play services now. Basically play services with no elevated perms that ship with stock Pixels. Sandboxed play has the same permissions as a normal android app, so you can still utilize FCM and other play features without sacrificing privacy

**Comment by tneotia** [comment]
Completed in full rewrite
